Tasks/AzureRmWebAppDeployment/operations/KuduServiceUtility.ts

```diff
@@ -0,0 +1,323 @@
+            tl.debug('creating kuduPostDeploymentScript_local file');
+            var scriptFilePath = path.join(tl.getVariable('AGENT.TEMPDIRECTORY'), isLinux ? 'kuduPostDeploymentScript_local.sh' : 'kuduPostDeploymentScript_local.cmd');
+            tl.writeFile(scriptFilePath, inlineScript);
+            tl.debug('Created temporary script file : ' + scriptFilePath);
+            return {
+                "filePath": scriptFilePath,
+                "isCreated": true
+            };
+        }
+        if(!tl.exist(scriptPath)) {
+            throw Error(tl.loc('ScriptFileNotFound', scriptPath));
+        }
+        var scriptExtension = path.extname(scriptPath);
+        if(isLinux){
+            if(scriptExtension != '.sh'){
+                throw Error(tl.loc('InvalidScriptFile', scriptPath));
+            }
+        } else {
+            if(scriptExtension != '.bat' && scriptExtension != '.cmd') {
+                throw Error(tl.loc('InvalidScriptFile', scriptPath));
+            }
+        }
+        tl.debug('postDeployment script path to execute : ' + scriptPath);
+        return {
+            filePath: scriptPath,
+            isCreated: false
+        }
+    }
+
+    private async _warFileDeployment(packagePath: string, physicalPath: string, virtualPath?: string): Promise<string> {
+        tl.debug('WAR: webAppPackage = ' + packagePath);
+        let warFile = path.basename(packagePath.slice(0, packagePath.length - '.war'.length));
+        let warExt = packagePath.slice(packagePath.length - '.war'.length)
+        tl.debug('WAR: warFile = ' + warFile);
+        warFile = warFile + ((virtualPath) ? "/" + virtualPath : "");
+        tl.debug('WAR: warFile = ' + warFile);
+        physicalPath = physicalPath + "/webapps/" + warFile;
+        await this.createPathIfRequired(physicalPath);
+        return physicalPath;
+
+    }
+
+    private async _appOfflineKuduService(physicalPath: string, enableFeature: boolean): Promise<void> {
+        if(enableFeature) {
+            tl.debug('Trying to enable app offline mode.');
+            var appOfflineFilePath = path.join(tl.getVariable('AGENT.TEMPDIRECTORY'), 'app_offline_temp.htm');
+            tl.writeFile(appOfflineFilePath, '<h1>App Service is offline.</h1>');
+            await this._appServiceKuduService.uploadFile(physicalPath, 'app_offline.htm', appOfflineFilePath);
+            tl.debug('App Offline mode enabled.');
+        }
+        else {
+            tl.debug('Trying to disable app offline mode.');
+            await this._appServiceKuduService.deleteFile(physicalPath, 'app_offline.htm');
+            tl.debug('App Offline mode disabled.');
+        }
+    }
+
+    private async _pollForFile(physicalPath: string, fileName: string, noOfRetry: number): Promise<void> {
+        var attempts: number = 0;
+        if(tl.getVariable('appservicedeploy.retrytimeout')) {
+            noOfRetry = Number(tl.getVariable('appservicedeploy.retrytimeout')) * 6;
```

Comment:
why retry timeout is being multiplied by 6 and how retry timeout changed to no. of retry ?

Reply:
we pass minutes to retry here. (30 minutes)
And, we convert it to number of retry: minutes*6 with a interval of 10 seconds.